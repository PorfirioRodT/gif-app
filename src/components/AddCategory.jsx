import { useState } from 'react';
import { getGifs } from '../helpers/getGifs';

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');    

    const onInputChange = ({target}) => {
        //console.log(event.target.value);
        setInputValue(target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();

        const trimmedInput = inputValue.trim();
        
        if(trimmedInput.length <= 1) return;

        //setCategories(categories => [inputValue, ...categories]);
        setInputValue('');
        onNewCategory(trimmedInput);
    }

    return (
        <form onSubmit={ onSubmit }>

            <input 
                type="text"
                placeholder="Buscar Gifs"
                value={ inputValue }
                onChange= { onInputChange }
            />

        </form>
    ) 
}
