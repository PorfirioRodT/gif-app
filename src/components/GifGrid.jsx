import { GifGridItem } from './GifGridItem';
import { UseFetchGifs } from '../hooks/UseFetchGifs';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {
  
    const {images, isLoading} = UseFetchGifs( category );

    return (
        <>
            <h3>{ category }</h3>

            {
                isLoading && (<h2>Cargando...</h2>)
            }

            <div className='card-grid'>
                { 
                    images.map( ( image ) => (
                        <GifGridItem 
                            key={ image.id }
                            { ...image }
                        />  
                    )) 
                }           
            </div>

        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}