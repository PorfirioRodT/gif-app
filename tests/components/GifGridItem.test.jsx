import { render, screen } from "@testing-library/react"
import { GifGridItem } from "../../src/components/GifGridItem";


describe('Tests in <GifGridItem />', () => { 

    const title = 'Apex';
    const url = 'https://apexlegends.com/wraithcalva.jpg';
    
    test('Should match with the snapshot', () => { 

        const { container } = render(<GifGridItem title={title} url={url} />);
        expect(container).toMatchSnapshot();

    });

    test('should show the image and with the url and the correct ALT', () => { 

        render(<GifGridItem title={ title } url={ url }/>);
        //screen.debug();
        //expect(screen.getByRole('img').src ).toBe(url);
        const {src, alt} = screen.getByRole('img');
        expect( src ).toBe(url);
        expect( alt ).toBe(alt);

    });

    test('should show the title in the component', () => { 

        render(<GifGridItem title={ title }/>)
        expect(screen.getByText(title)).toBeTruthy();

    });

})