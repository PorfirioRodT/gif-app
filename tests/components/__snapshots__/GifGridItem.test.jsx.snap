// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifGridItem /> Should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Apex"
      src="https://apexlegends.com/wraithcalva.jpg"
    />
    <p>
      Apex
    </p>
  </div>
</div>
`;
